drivers/can/can_mcan.c

```diff
@@ -76,6 +48,27 @@ int can_mcan_write_reg(const struct device *dev, uint16_t reg, uint32_t val)
 	return err;
 }
 
+int can_mcan_read_mram(const struct device *dev, uint16_t offset, void *dst, size_t len)
```

Comment:
Could these functions be static?

Reply:
They cannot, unfortunately. They are exposed in the `can_mcan.h` header for use in out-of-tree driver implementations, where the frontend driver (may) need to access Message RAM directly. One could argue that the frontend driver could just use its private MRAM access functions, but the parameters there are different.

It's just one function call deeper by not inlining, I think it is okay.

Reply:
I agree, that sounds reasonable